crates/services/upgradable-executor/src/executor.rs

```diff
@@ -0,0 +1,280 @@
+use crate::config::Config;
+use fuel_core_executor::{
+    executor::{
+        ExecutionBlockWithSource,
+        ExecutionOptions,
+        OnceTransactionsSource,
+    },
+    ports::{
+        RelayerPort,
+        TransactionsSource,
+    },
+};
+use fuel_core_storage::{
+    column::Column,
+    kv_store::KeyValueInspect,
+    transactional::{
+        AtomicView,
+        Changes,
+        Modifiable,
+    },
+};
+#[cfg(any(test, feature = "test-helpers"))]
+use fuel_core_types::services::executor::UncommittedResult;
+use fuel_core_types::{
+    blockchain::primitives::DaBlockHeight,
+    fuel_tx::Transaction,
+    fuel_types::BlockHeight,
+    services::{
+        block_producer::Components,
+        executor::{
+            ExecutionResult,
+            ExecutionTypes,
+            Result as ExecutorResult,
+            TransactionExecutionStatus,
+        },
+        Uncommitted,
+    },
+};
+use std::sync::Arc;
+
+/// The upgradable executor supports the WASM version of the state transition function.
+/// If the block has a version the same as a native executor, we will use it.
+/// If not, the WASM version of the state transition function will be used
+/// (if the database has a corresponding bytecode).
+pub struct Executor<S, R> {
+    pub storage_view_provider: S,
+    pub relayer_view_provider: R,
+    pub config: Arc<Config>,
+    #[cfg(feature = "wasm-executor")]
+    engine: wasmtime::Engine,
+    #[cfg(feature = "wasm-executor")]
+    module: wasmtime::Module,
+}
+
+impl<S, R> Executor<S, R> {
+    pub fn new(
+        storage_view_provider: S,
+        relayer_view_provider: R,
+        config: Arc<Config>,
```

Comment:
It looks like all instances of `new` create the `Arc` inline. I might suggest taking `config` as `config: Config` and  doing:
```
Self {
   ...
   config: Arc::new(config)
}
```

to simplify things a little.